File: src/limelight/ui/ButtonGroupCache.java
//- Copyright © 2008-2010 8th Light, Inc. All Rights Reserved.
//- Limelight and all included source files are distributed under terms of the GNU LGPL.

package limelight.ui;

import java.util.Hashtable;

public class ButtonGroupCache
{
  private final Hashtable<String, RadioButtonGroup> cache;

  public ButtonGroupCache()
  {
    cache = new Hashtable<String, RadioButtonGroup>();
  }

  public RadioButtonGroup get(String groupName)
  {
    if(!cache.containsKey(groupName))
      cache.put(groupName, new RadioButtonGroup());
    return cache.get(groupName);
  }

  public Hashtable<String, RadioButtonGroup> getCache()
  {
    return cache;
  }
}

File: lib/limelight/button_group_cache.rb
#- Copyright © 2008-2010 8th Light, Inc. All Rights Reserved.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

module Limelight
  ButtonGroupCache = UI::ButtonGroupCache #:nodoc:
  class ButtonGroupCache #:nodoc:
    def [](key)
      return self.get(key)
    end
  end
end
